GG/GG/Font.h

```diff
@@ -147,6 +151,13 @@ class GG_API Font
             pair.second. */
         Substring(const std::string& str_, const IterPair& pair);
 
+        /** Attach this Substring to \p str_.
+
+            This changes the iterators from pointing into the previous
```

Comment:
It is the same behavior as the C++ standard library.  

When an operation on a container, insert,  delete etc., invalidates the iterators for that container, the iterators themselves are not changed, but they are invalid by definition and using them has undefined behavior. 

In our case, once the `Substring` has been bound to a new `std::string` any iterators stored are invalid.  They might by accident produce correct results if the old string and the new string are identical, but they are definitely pointing to the a string not bound to this `Substring`and perhaps a string that no longer exists.